src/pages/SignUpForm/SignUpForm.tsx

```diff
@@ -24,17 +25,41 @@ const schema = Yup.object().shape({
 
 const SingUpForm: React.FC = () => {
 
-    const listOfCurrencies = ["1", "2", "3"]
-
+    const [currency, setCurrency] = React.useState("DOLLAR");
     const [showPassword, setShowPassword] = React.useState(false);
+    const [listOfCurrencies, setListOfCurrencies] = React.useState([]);
+    const [register, setRegistre] = React.useState(false);
+    const [errorRegister, setErrorRegistre] = React.useState(false);
+    const [errorMessage,setErrorMessage] = React.useState('')
+
     const handleClickShowPassword = () => setShowPassword((show) => !show);
     const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
         event.preventDefault();
     };
     const handleSignUp = (values: any) => {
-        // Handle form submission or API call here
-        console.log(values);
+        signUp(values.fullName, values.companyName, currency, values.email, values.password)
+            .then(response => {
+                setRegistre(true);
+                setErrorRegistre(false);
+            })
+            .catch(error => {
+                setErrorRegistre(true);
+
+                if (error.code == 'ERR_BAD_REQUEST') {
+                    setErrorMessage('email or company name already exists')
+                }
+                else {
+                    setErrorMessage('the request could not be completed, please try again')
+                }
+
+            })
     };
+    const getCurrencies1 = async () => {
```

Comment:
change to:
    useEffect(() => {
         const getCurrenciesAsync = async () => {
               const res= await getCurrencies();
               setListOfCurrencies(res.data)
           }
        getCurrenciesAsync();
    }, []);